src/impl/realm_coordinator.cpp

```diff
@@ -88,7 +89,8 @@ void RealmCoordinator::create_sync_session()
         if (auto self = weak_self.lock()) {
             if (self->m_transaction_callback)
                 self->m_transaction_callback(old_version, new_version);
-            self->notify_others();
+            if (self->m_notifier)
```

Comment:
While `wake_up_notifier_worker()` does the same thing as this, it actually shouldn't and I've added a FIXME there to reflect that.